### ISSUE TITLE
[SearchK()] Result Issue

### ISSUE THREAD

**Issue by iamkbpark** [bstewart/stm#114]
Dear Dr. Stewart:

I tested and it worked with the fixed master version.
However, the resulting diagnostic values differ significantly for Held-Out Likelihood and Residuals
Those two values from 1.3 version are trending exact opposite for the ones from 1.2.
Could you advise what might cause this?

If you need my data object and codes, I will be more than happy to share with you.
Then, please let me know how I can send you my codes and data object.
Thank you for your time and help.

Sincerely,
KB Park

**Comment by bstewart** [comment]
Could you try running it again a few times with different seeds?  The calculations in searchK are based on held out samples which particularly for small samples can be relatively unstable.  I ran a few examples under v1.2 and under v1.3 with the Gadarian data and could not find any indication that there was greater instability under v1.3.  Running the demo in the searchK help file I couldn't get any instance where the trend exactly flipped, but i could definitely get some cases where the trend changed.  For example- below I have included two of the most distinct plots I got on 5-6 runs. That said- this should stabilize in settings where the documents are reasonably long and there are enough documents.  I'll add an issue to clarify this in the documentation
![5bd9c480-8f30-4e76-8ee1-875118158062](https://user-images.githubusercontent.com/577337/31126799-cedc68b8-a81a-11e7-832b-3f9977de6176.png)
![93abb05b-d954-45d0-bff6-fc883501baba](https://user-images.githubusercontent.com/577337/31126821-e1b34aa6-a81a-11e7-9d00-09905190ac24.png)

**Comment by iamkbpark** [comment]
Dear Dr. Stewart:

Thank you for the explanations. 
So, I tried running **`searchK()`** five times each with v1.2 and v1.3(cores=2).
The corpus(online review) used for both has 5,461 docs, 1,412 terms, and 87,471 tokens.

**[v1.2] Plot**
Though there is slight difference among the runs, I got similar patterns for all four plots as below:
![1 2_1003-2](https://user-images.githubusercontent.com/5588007/31134501-41bcdad4-a89d-11e7-8a70-6368da439f0b.png)

**[v1.3] Plot **
I got the similar patterns for all four plots, but different with v1.2 for Held-Out and Residual plots
![1 3-core2_1003-3](https://user-images.githubusercontent.com/5588007/31134661-bd4d915c-a89d-11e7-882e-002a71420f93.png)

Did I do something wrong or miss the point?
Could you advise?
Your advice will be greatly appreciated.

Sincerely,
KB Park

**Comment by bstewart** [comment]
Yeah that seems pretty strange.  Could you send me replication data?  I'll test as soon as I can.

**Comment by iamkbpark** [comment]
Dear Dr. Stewart: 
Thank you for your comment. 
I could hardly find your email here in Github(my first time..), so I wonder if it is ok to email you at princeton address? Thank you for your time. -Sincerely, KB Park

**Comment by bstewart** [comment]
Yes- that would be great. bms4@princeton.edu

**Comment by iamkbpark** [comment]
I just emailed you. Thank you for your time!

**Comment by bstewart** [comment]
Sorry for the long delay on this: I got back to working on it.  So I'm having a hard time replicating your v1.3 findings.  Using your data here are two plots:

v1.3.1 (current Github release)
![image](https://user-images.githubusercontent.com/577337/31854855-d55999d4-b66d-11e7-8556-fa3dcfa11401.png)

v.1.2.2 (old CRAN release)
![image](https://user-images.githubusercontent.com/577337/31854977-36fb1702-b66f-11e7-8789-600970735503.png)

Note these are substantially the same and basically both look like your V1.2 pattern.  I got a bit lazy and didn't do the full set for the v1.2.2 graphic because I didn't have multicore but you can see the same pattern there anyway.

Could you perhaps provide the exact replication code to get your V1.3 finding?  I just did:
```
load("doc_vocab_meta.Rda")
sk <- searchK(documents=docs, vocab=vocab, K=5:15)
plot(sk)
```

**Comment by iamkbpark** [comment]
Dear Dr. Stewart:  Thank you so much to investigate further on this issue.  The original code I ran was:

`sk3 <- searchK (cores = 2,  K = 5:15, init.type = 'Spectral', ngroups = 4, prevalence = ~ bs(RATING_DATE), 
 documents = docs, vocab = vocab, data = meta, interactions = F, max.em.its = 500)`

I tested several `searchk()` fits both on MAC(cores=2) and Window, and found out that the problem I had occurred when 'ngroups=4' was set in vs1.3.1.
Is this option not recommended or allowed when running `searchk()` in vs1.3.1?

Especially but importantly, in MAC with vs1.3.1, it crashes with above codes: without ngroup

Other than that, everything works perfect with vs1.3.1 and I will also email you for the test I did for your reference.
I do appreciate your help.

**Comment by bstewart** [comment]
I honestly have no idea what's going on here. The searchK multicore functionality was added as a pullrequest but at this point I may just pull it out because it is too inconsistent.  I'll work on it for a while.

**Comment by iamkbpark** [comment]
Sure! Thanks again for your help

**Comment by bstewart** [comment]
I looked at this again today- I have to admit I'm a bit baffled.  I can now get the above code to replicate if you switch bs(RATING_DATE) to s(RATING_DATE)  or by explicitly loading the splines package first.

I still need to hunt down a Windows test machine to evaluate on there- but it seems to work fine on the Mac as best I can tell over here.
